### ISSUE TITLE
Windows installation guide links to download Visual Studio 2022 but Boost supports up to 2019

### ISSUE THREAD

**Issue by Synful-Syn** [azerothcore/wiki#678]
**Describe the bug**
The requirements of the [Windows installation guide](https://www.azerothcore.org/wiki/windows-requirements) ([git link](https://github.com/azerothcore/wiki/edit/master//docs/windows-requirements.md)) asks to install `Visual Studio (16) Community 2019` but the download link leads to 2022 and its default MSVC is v143 but Boost (1.78, the latest) supports up to v142.

On the linked page, it does mention about 2019 but it is a download link for Mac.

**To reproduce**
Pretty much follow the guide on a clean Windows 10 but I choose Boost 1.78 because I could.

**Expected behavior**
A clear and concise description of what I should do without doing any extra steps in order to achieve a working local AzerothCore on my computer.

**Screenshots**
View the **Additional context** part.

**Desktop:**
 - Windows 10 x64
 - Firefox

**Additional context**
TLDR: I simply had a new computer so I am re-doing the setup and this is my analysis on how to I managed to compile with Visual Studio by saying "compile the way Visual Studio 2019 would do".

Fast forward in the [core installation](https://www.azerothcore.org/wiki/windows-core-installation), in Visual Studio 2022, I did "Regenerate" on ALL_BUILDS in the `Solution Explorer`, I then I had these errors.
(impossible to open the file `boostlib_filesystem-vc143-mt-x64-1_78.lib`)
![image](https://user-images.githubusercontent.com/22718174/148718161-711bdd39-a4e3-4715-82c2-aef856df7a80.png)

I've noticed the error messages were very similar to what I've previously seen in CMake. However, the error was looking for v143 while Boost, the latest version available is v142.
![image](https://user-images.githubusercontent.com/22718174/148793610-1323b5bd-cb54-4e3a-87d9-a765c2267430.png)

Luckily, I did remember seeing that by default, Visual Studio 2022 Community installs MSCV v143 and I have chosen to modify my Visual Studio so I would also have the v142.
![image](https://user-images.githubusercontent.com/22718174/148793984-685c0b41-277e-464c-ad7b-df08d49a4e24.png)

It did not fix my issue, but I've understood it was still trying to use MSVC v143 by default. After some extensive research, I've found out that in the `Solution Explorer`, you can get into the properties of a solution and tell it to use v142 instead and I tried on one of the projects: one of the 2 messages were gone so I knew if all of them were on v142, it would finally be fixed.
![image](https://user-images.githubusercontent.com/22718174/148720276-9bd21859-ae7d-4793-b496-db8955c56c29.png)
![image](https://user-images.githubusercontent.com/22718174/148719665-50fcab5f-aa52-47d2-80b1-8d959cb9533a.png)

**Comment by Kitzunu** [comment]
![image](https://user-images.githubusercontent.com/24550914/149598221-cc5c2d9a-15f9-487d-81c5-48774f8d24de.png)

Don't use Boost 1.78

**Comment by Winfidonarleyan** [comment]
Rofl, just look this

![image](https://user-images.githubusercontent.com/18329778/149598622-36190268-dafb-48ad-8bc4-18fdef75c70c.png)

![image](https://user-images.githubusercontent.com/18329778/149598636-6d430296-2333-42c6-bef1-c7ba08c6e573.png)

**Comment by Winfidonarleyan** [comment]
@Synful-Syn for v143 need `lib64-msvc-14.3`

**Comment by Winfidonarleyan** [comment]
https://boostorg.jfrog.io/artifactory/main/release/1.78.0/binaries/boost_1_78_0-msvc-14.3-64.exe

**Comment by Synful-Syn** [comment]
Oh, thank you for the link, I don't know how I missed it!

I still think the Windows installation guide is still missing a few points.
As I firstly mentionned, the download link for `Visual Studio (16) Community 2019` leads to 2022 and **I cannot find where to download 2019** (but I still have managed to get my server up and running using Visual Studio 2022 compiled with MSVC 143 and the 1.78 MSVC 143 Boost link above).

For the second, let's begin with how the page layout currently is.
![image](https://user-images.githubusercontent.com/22718174/149607199-ef59c50b-eec6-4415-a156-39d331e213eb.png)

On point `i`, it says `Download the prebuilt Windows Binary for Visual Studio 2019`. What is it? Is it the link on either point `iii` or point `iv` or is it something I can find in the Visual Studio Installer?

On points `iii` and `iv`, they leads to a MSVC 142 but as I said, the Visual Studio download link leads to a version that has MSVC 143 which is a mismatch.

**I propose to change the layout about the Boost section to be the following.**
![image](https://user-images.githubusercontent.com/22718174/149606088-3f159301-3a4e-40b7-af2b-cc6d9d9f6dc0.png)
The website link has been moved after the recommended download links in order to prevent the situation of missing the recommended download links by clicking the very first link available.
The recommended download links also shows which link supports which version of Visual Studio between 2019 and 2022.

**Comment by Winfidonarleyan** [comment]
@Synful-Syn `lib64-msvc-14.3` only in boost `1.78.0`
![image](https://user-images.githubusercontent.com/18329778/149607679-51e283fe-fa48-4986-8e17-03a7d7dad555.png)

**Comment by Winfidonarleyan** [comment]
After start cmake support boost `1.78.0` we can change guide

**Comment by Winfidonarleyan** [comment]
And we can start make PR
